packages/grafana-ui/src/components/DeleteButton/DeleteButton.tsx

```diff
@@ -32,34 +43,74 @@ export class DeleteButton extends PureComponent<Props, State> {
     });
   };
 
+  getStyles = () => {
+    return {
+      container: css`
+        width: 24px;
+        direction: rtl;
+        display: flex;
+        align-items: center;
+      `,
+      deleteButton: css`
+        position: absolute;
+        opacity: 0;
+        transition: opacity 0.1s ease;
+        z-index: 0;
+      `,
+      deleteButtonVisible: css`
+        opacity: 1;
+        transition: opacity 0.1s ease;
+        z-index: 2;
+      `,
+      confirmDelete: css`
+        display: flex;
+        align-items: flex-start;
+        opacity: 0;
+        transition: opacity 0.12s ease-in, transform 0.14s ease-in;
+        transform: translateX(100px);
+      `,
+      confirmDeleteShow: css`
+        opacity: 1;
+        transition: opacity 0.08s ease-out, transform 0.1s ease-out;
+        transform: translateX(0);
+      `,
+    };
+  };
+
   render() {
     const { onConfirm } = this.props;
-    let showConfirm;
-    let showDeleteButton;
-
-    if (this.state.showConfirm) {
-      showConfirm = 'show';
-      showDeleteButton = 'hide';
-    } else {
-      showConfirm = 'hide';
-      showDeleteButton = 'show';
-    }
+    const styles = this.getStyles();
 
     return (
-      <span className="delete-button-container">
-        <a className={'delete-button ' + showDeleteButton + ' btn btn-danger btn-small'} onClick={this.onClickDelete}>
+      <span css={styles.container}>
+        <a
+          css={css`
+            ${styles.deleteButton};
+            ${!this.state.showConfirm ? styles.deleteButtonVisible : ''};
+          `}
+          className="btn btn-danger btn-small"
```

Comment:
`css` / `className` composition enables us to use both "legacy" and newly created styles